.codecov.yml

```diff
@@ -0,0 +1,25 @@
+# -- repository yaml --
+
+# Explicitly wait for all jobs to finish, as wait_for_ci prematurely triggers.
+# See https://github.com/python-trio/trio/issues/2689
+codecov:
+  notify:
+    # This number needs to be changed whenever the number of runs in CI is changed.
+    # Another option is codecov-cli: https://github.com/codecov/codecov-cli#send-notifications
+    after_n_builds: 78
+    wait_for_ci: false
+    notify_error: true # if uploads fail, replace cov comment with a comment with errors.
```

Comment:
I think it makes sense to also set `fail-on-error` in the action invocations. It's false by default, resulting in hidden failures in the job logs.